pkg/stanza/fileconsumer/internal/reader/reader.go

```diff
@@ -164,7 +164,7 @@ func (r *Reader) Close() *Metadata {
 func (r *Reader) Read(dst []byte) (int, error) {
 	// Skip if fingerprint is already built
 	// or if fingerprint is behind Offset
-	if len(r.Fingerprint.FirstBytes) == r.FingerprintSize || int(r.Offset) > len(r.Fingerprint.FirstBytes) {
+	if r.Fingerprint.BytesLength == r.FingerprintSize || int(r.Offset) > r.Fingerprint.BytesLength {
```

Comment:
Can we create a method to encapsulate this logic? `IsCompleteFingerPring(maxFingerPrintSize, Offset)`

Reply:
This should be in fingerprint class?

Reply:
correct, in the FingerPrint struct. it is part of the fingerprint logic.